File: SMarketPlace/SMarketPlace.Web/Modules/Sucursales/TblSucursal/TblSucursalForm.cs
using Serenity.ComponentModel;

namespace SMarketPlace.Sucursales.Forms;

[FormScript("Sucursales.TblSucursal")]
[BasedOnRow(typeof(TblSucursalRow), CheckNames = true)]
public class TblSucursalForm
{
    public string LocalSap { get; set; }
    public string NombreSuc { get; set; }
    public string Direccion { get; set; }
    public string Codigopostal { get; set; }
    public int Pais { get; set; }
    public int Ciudad { get; set; }
    public int Estado { get; set; }
    public int Region { get; set; }
    public int Zona { get; set; }
}

File: SMarketPlace/SMarketPlace.Web/Modules/ServerTypes/Sucursales/TblSucursalForm.ts
import { StringEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface TblSucursalForm {
    LocalSap: StringEditor;
    NombreSuc: StringEditor;
    Direccion: StringEditor;
    Codigopostal: StringEditor;
    Pais: IntegerEditor;
    Ciudad: IntegerEditor;
    Estado: IntegerEditor;
    Region: IntegerEditor;
    Zona: IntegerEditor;
}

export class TblSucursalForm extends PrefixedContext {
    static formKey = 'Sucursales.TblSucursal';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TblSucursalForm.init)  {
            TblSucursalForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(TblSucursalForm, [
                'LocalSap', w0,
                'NombreSuc', w0,
                'Direccion', w0,
                'Codigopostal', w0,
                'Pais', w1,
                'Ciudad', w1,
                'Estado', w1,
                'Region', w1,
                'Zona', w1
            ]);
        }
    }
}